tests/test_create_employee.py

```diff
@@ -50,5 +48,51 @@ def test_create_employee_no_creds(self):
         browser.find_element(By.ID, "btnSave").click()  # Save button
 
         browser.find_element(By.LINK_TEXT, "PIM").click()
-        pass
 
+    def test_validate_employee_list_sorting_rules(self):
+        browser = self.browser
+        wait = WebDriverWait(browser, 5)
+
+        # Step 1: Open Employee List page
+        browser.find_element(By.ID, "menu_pim_viewEmployeeList").click()
+        wait.until(expected_conditions.url_contains("/pim/viewEmployeeList"))
+        self.assertEqual("Employee List", browser.find_element(By.CLASS_NAME, "selected").text)
+
+        # Step 2: Scroll down to the Employee list table
+        browser.find_element(By.ID, 'footer').location_once_scrolled_into_view
+
+        # Step 3. Click the "First (Middle) Name" table header
+        browser.find_element(By.LINK_TEXT, "First (& Middle) Name").click()
+
+        # 4. Wait for the sort to complete (hint: url)
+        wait.until(expected_conditions.url_contains("ASC"))
+        url_partial = browser.current_url.split("?")
+        self.assertEqual("sortField=firstMiddleName&sortOrder=ASC", url_partial[1])
+
+        # 5. Ensure the records on the first page listed in alphabetical order by name
+        rows = browser.find_elements(By.XPATH, "//tbody/tr")
+        list_of_names = []
+
+        for single_row in rows:
+            list_of_names.append(single_row.find_element(By.XPATH, ".//td[3]").text)
+
+        print(list_of_names)
+        self.assertEqual(sorted(list_of_names), list_of_names)
+
+        # Bonus Challenge
+        form = browser.find_element(By.ID, "frmList_ohrmListComponent")
+        pages = form.find_elements(By.XPATH, "//div[1]/ul")
```

Comment:
Unfortunately this would only work when there are 5 or less page, after that it does not display the last page number, so if there are 13  pages, you would still only see page 5 as the last number and then ...

Reply:
Thank you for your review. Added bunch of code to extract the total number of records and to compare with the current page max number (1-50 of 74) aka 50 vs 74. Not the most elegant solution, but it works.